### PR TITLE
feat/AB#73441_build_custom_expression_builder

### PR DESCRIPTION
**Pull request by RenzoPrats**
# Description

Created CronEditor component.

## Useful links

[ticket](https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/73441)
[reference](https://github.com/haavardj/ngx-cron-editor/blob/master/libs/ngx-cron-editor/src/enums.ts)

## Type of change

- [X] Improvement (refactor or addition to existing functionality)

# How Has This Been Tested?

It has been tested verifying in the storybook if the new component loads correctly and in the application in enter a schedule when editing or adding a pull job if the component is working properly.

## Screenshots

Peek 28-08-2023 18-01

# Checklist:

( * == Mandatory ) 

- [X] * I have set myself as assignee of the pull request
- [X] * My code follows the style guidelines of this project
- [X] * Linting does not generate new warnings
- [X] * I have performed a self-review of my own code
- [X] * I have put the ticket for review, adding the oort-frontend team to the list of reviewers
- [X] * I have commented my code, particularly in hard-to-understand areas
- [X] * I have put JSDoc comment in all required places
- [ ] * My changes generate no new warnings
- [X] * I have included screenshots describing my changes if relevant
- [X] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/ui/src/lib/cron-editor/cron-editor.component.html [hunk 1]
```diff
@@ -0,0 +1,310 @@
+<section class="cron-editor-main">
+    <mat-tab-group class="cron-editor-container" (selectedTabChange)="onTabChange($event)" [backgroundColor]="backgroundColor" [color]="color">
```
Comment (AntoineRelief) [inline libs/ui/src/lib/cron-editor/cron-editor.component.html]: @RenzoPrats 
I know this is the first commit, but just to make sure it was well understood:
- we should use our new UI components, not material
- css must be written using our new tailwind options,

**Comment by NathanHGit** [comment]
Everything works fine but sometimes the display looks a bit confusing. 

Here is a suggestion for a new display:

peek

And here are some ideas of improvements:
1. set the default value to the corresponding fields when opening the pop up (if they are not null) and switch to the correct tab
2. when changing tab:
     - do not reset the top text or
     - reset past tab fields or
     - set top text with new tab fields (if not null)
5. set the fields as required in the current tab or indicate that some fields are missing to save the format

**Comment by NathanHGit** [comment]
I refactored the code and made some changes: 
- I changed inline-flex to flex in order to use align-items property
- I used gap instead of margins
- I added a scss property to prevent the bold effect from being applied to all radio button labels  
- I added spans with colon to the time-picker component to make it more aesthetic 
- I added a padding top to the cron-editor component

 @RenzoPrats you changed the ui here but it might be more confusing than before (every 'hours' at time 'minutes')

image

You also added the current value when opening the pop-up but that seems to work only for minutes and hours inputs, is it relevant that way ?

note: it seems we don't have the same code formatter configuration

**Comment by AntoineRelief** [comment]
@RenzoPrats I put it back as draft
please remove the draft when okay

File: libs/ui/src/lib/cron-editor/cron-editor.component.html [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+      <!-- Advanced-->
+      <ui-tab class="cron-editor-tab" *ngIf="!options.hideAdvancedTab" #advancedTab>
+        <ng-container ngProjectAs="label">
+          <label>Advanced</label>
+        </ng-container>
+        <ng-template uiTabContent>
+          <div class="cron-editor-tab-content" [formGroup]="allForm">
+            <div uiFormFieldDirective>
+              <label>Expression</label>
+              <input matInput type="text" class="advanced-cron-editor-input" formControlName="expression">
```
Comment (AntoineRelief) [inline libs/ui/src/lib/cron-editor/cron-editor.component.html]: @RenzoPrats 
Doing the review after quite some time
matInput is banned !!!
we have tailwind

File: libs/ui/src/lib/cron-editor/cron-editor.component.html [hunk 3]
```diff
@@ -0,0 +1,415 @@
...
+        </ng-template>
+      </ui-tab>
+  
+      <!-- Advanced-->
+      <ui-tab class="cron-editor-tab" *ngIf="!options.hideAdvancedTab" #advancedTab>
+        <ng-container ngProjectAs="label">
+          <label>Advanced</label>
+        </ng-container>
+        <ng-template uiTabContent>
+          <div class="cron-editor-tab-content" [formGroup]="allForm">
```
Comment (AntoineRelief) [inline libs/ui/src/lib/cron-editor/cron-editor.component.html]: @RenzoPrats 
stop using classes when it does not worth it
there's a lot to improve about styling there
most classes are useless, many are just having one property, and in some cases, they don't apply at all
use tailwind !!

**Pull request merged**